### PR TITLE
Setup engine

### PR DESCRIPTION
**Pull request by porqueoutai**
Setup of engine on a fresh installed ubuntu 18.04.
Had some compile errors as described in commits.

### CODE REVIEW

File: docs/Dependencies.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Dependencies
 * cmake
+* nightly-build
```
Comment (mgerhardy) [inline docs/Dependencies.md]: What's this about? `apt-cache show nightly-build` doesn't return anything on my system (ubuntu specific?).

File: docs/Dependencies.md [hunk 2]
```diff
@@ -13,6 +14,7 @@
   * gtest
   * opencl
   * libuuid
+  * mosquitto
```
Comment (mgerhardy) [inline docs/Dependencies.md]: should maybe get marked as (optional) - as it's only needed to compile the Traze Client (https://github.com/iteratec/traze)

File: docs/Dependencies.md (hunk 1 above)
Comment (porqueoutai) [reply docs/Dependencies.md]: Yea, was the wrong one, it was ninja-build. Don't know how nightly came in my mind.

File: docs/Dependencies.md (hunk 2 above)
Comment (porqueoutai) [reply docs/Dependencies.md]: Removed the package install and marked it as optional.

**Pull request merged**